### PR TITLE
[enh] Check domain expiration date

### PR DESCRIPTION
**Pull request by zamentur**
## The problem
Sometimes user could forgot to renew their domain

## Solution

Check if the date is soon with whois (error if <=7days, warning if <=30 days or info)

expirations

## PR Status

Ready

## How to test

apt install whois

## Validation

- [ ] Principle agreement 0/2 : 
- [ ] Quick review 0/1 : 
- [ ] Simple test 0/1 : 
- [ ] Deep review 0/1 :

### CODE REVIEW

**Comment by zamentur** [comment]
I have tested it on more than 40 domains.

Now i am blacklisted: # Your IP has been restricted due to excessive access, please wait a bit

I think it could be reviewed and merge

File: data/hooks/diagnosis/12-dnsrecords.py [hunk 1]
```diff
@@ -128,14 +139,106 @@ def current_record_match_expected(self, r):
...
+        for line in out:
+            match = re.search(r'Expir.+(\d{4}-\d{2}-\d{2})', line, re.IGNORECASE)
+            if match is not None:
+                return datetime.strptime(match.group(1), '%Y-%m-%d')
+
+            match = re.search(r'Expir.+(\d{2}-\w{3}-\d{4})', line, re.IGNORECASE)
+            if match is not None:
+                return datetime.strptime(match.group(1), '%d-%b-%Y')
+
+        return "expiration_not_found"
```
Comment (alexAubin) [inline data/hooks/diagnosis/12-dnsrecords.py:240]:
```suggestion
        return "not_found"
```

File: data/hooks/diagnosis/12-dnsrecords.py [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
+from publicsuffix import PublicSuffixList
 
 from moulinette.utils.filesystem import read_file
+from moulinette.utils.process import check_output
 
 from yunohost.utils.network import dig
 from yunohost.diagnosis import Diagnoser
 from yunohost.domain import domain_list, _build_dns_conf, _get_maindomain
 
+PENDING_SUFFIX_LIST = ['ynh.fr', 'netlib.re']
```
Comment (kay0u) [inline data/hooks/diagnosis/12-dnsrecords.py:17]:
```suggestion
PENDING_SUFFIX_LIST = ['ynh.fr', 'netlib.re', 'nohost.me', 'noho.st']
```

Comment (zamentur) [reply data/hooks/diagnosis/12-dnsrecords.py:17]: nohost.me and noho.st is already in public suffix list

**Comment by zamentur** [comment]
I just remove the not_working detection cause whois on .com exit with code 1 and other tested tld with 0 ...

**Pull request merged**